JavaScript/4-newfunc.js

```diff
@@ -0,0 +1,78 @@
+'use strict';
+
+function Node(parent, name, data) {
+  this.name = name;
+  this.data = data;
+  if (parent) {
+    this.parent = parent;
+    parent[name] = this;
+  }
+  this.child = [];
+}
+
+function Tree(name, data) {
+  this.root = new Node(null, name, data);
+}
+
+Tree.prototype.visitDepth = function(callback) {
+
+  (function recursive(currNode) {
+
+    const len = currNode.child.length;
+    let num;
+    for (num = 0; num < len; ++num) {
+      recursive(currNode.child[num]);
+    }
+
+    callback(currNode);
+  })(this.root);
+};
+
+Tree.prototype.isHave = function(callback) {
```

Comment:
`Tree.has(node)` returns boolean, no need callback but where is node argument to check?